core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -232,5 +235,9 @@ class CityInfo {
         }
         return false
     }
+
+    fun canAttack(): Boolean {
```

Comment:
This function can be avoided if can Attack replaces attacksThisTurn as a parameter :)